lib/util/chat/pages/chat_page.dart

```diff
@@ -0,0 +1,119 @@
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+class ChatPage extends StatefulWidget {
+  const ChatPage({required this.rideId, required this.profile, super.key});
+  final int rideId;
+  final Profile profile;
+
+  @override
+  State<ChatPage> createState() => _ChatPageState();
+}
+
+class _ChatPageState extends State<ChatPage> {
+  late Stream<List<Message>> _messagesStream;
+
+  @override
+  void initState() {
+    _messagesStream = SupabaseManager.supabaseClient
+        .from('messages')
+        .stream(primaryKey: ['id'])
+        .eq('ride_id', widget.rideId)
+        .order('created_at')
+        .map((messages) => Message.fromJsonList(messages));
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _messagesStream.drain();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: ProfileWidget(
+          widget.profile,
+          isTappable: true,
+        ),
+      ),
+      body: StreamBuilder<List<Message>>(
+        stream: _messagesStream,
+        builder: (context, snapshot) {
+          if (snapshot.hasData) {
+            List<Message> messages = snapshot.data!;
+            return Column(
+              children: [
+                Expanded(
+                  child: messages.isEmpty
+                      ? Column(
+                          mainAxisAlignment: MainAxisAlignment.center,
+                          children: [
+                            Image.asset(
+                              'assets/chat_shrug.png',
+                              scale: 8,
+                            ),
+                            const SizedBox(height: 16),
+                            Text(
+                              S.of(context).pageChatEmptyTitle,
+                              style: Theme.of(context).textTheme.headline6,
+                            ),
+                            const SizedBox(height: 8),
+                            Text(
+                              S.of(context).pageChatEmptyMessage,
+                            ),
+                          ],
+                        )
+                      : ListView(
+                          reverse: true,
+                          children: _buildChatBubbles(messages, widget.profile),
+                        ),
+                ),
+                MessageBar(widget.rideId),
+              ],
+            );
+          } else {
+            return const Center(child: CircularProgressIndicator());
+          }
+        },
+      ),
+    );
+  }
+
+  List<Widget> _buildChatBubbles(List<Message> messages, Profile? otherProfile) {
+    List<Widget> chatBubbles = [];
+    Profile ownProfile = SupabaseManager.getCurrentProfile()!;
+    Message currentMessage = messages.first;
+    bool isOwnMessage = currentMessage.senderId == ownProfile.id;
+    if (!isOwnMessage) currentMessage.markAsRead();
+    chatBubbles.add(ChatBubble(
```

Comment:
wouldn't it be easier to pass the whole message to ChatBubble? would remove the need for most of these arguments

Reply:
not sure, but if you want i can move this all to the ChatBubble

Reply:
What do you mean by "all this"? I just want the ChatBubble to have the "message" and to remove the "text", "time", ... arguments

Reply:
i'll do it